### DISCUSSION TITLE
How can I get all values of dirtyfield?

### DISCUSSION THREAD

**Discussion by goffxnca** [react-hook-form/react-hook-form#3275] [Q&A] [answered]
Now dirty field seem like it support only for check if which fields is dirty but not it value?

**Comment by bluebill1049** [comment]
Could you please share a codesandbox?

**Comment by jordanwallwork** [comment, marked as answer]
Could you do something like the following?

    const { register, handleSubmit, formState, getValues } = useForm();
    const dirtyFields = Object.keys(formState.dirtyFields).map((x) => ({
      field: x,
      value: getValues(x)
    }));`

https://codesandbox.io/s/dirty-field-values-hqg3j?file=/src/index.js

**Reply by goffxnca** [reply]
> Could you do something like the following?
> 
> ```
> const { register, handleSubmit, formState, getValues } = useForm();
> const dirtyFields = Object.keys(formState.dirtyFields).map((x) => ({
>   field: x,
>   value: getValues(x)
> }));`
> ```
> 
> https://codesandbox.io/s/dirty-field-values-hqg3j?file=/src/index.js

Wow, That is what I am looking for Thanks !!

**Reply by jonybekov** [reply]
What if it's deeply nested form?

**Reply by stychu** [reply]
Exactly. I've hit a wall upon a deeply nested form rn. Im very surprised there is no API level functionality to get the values of dirtyFields when RHF know what fields changed already. Why not include values instead just boolean flag in dirtyFields?

**Reply by uttampun44** [reply]
it works for me also
